kore/test/Test/Kore/Step/Simplification/And.hs

```diff
@@ -406,11 +408,133 @@ test_andSimplification =
                     , substitution = mempty
                     }
                 Conditional
-                    { term = Mock.constr10 gOfX
+                    { term = Mock.constr10 fOfX
                     , predicate = mkNot <$> makeCeilPredicate_ fOfX
                     , substitution = mempty
                     }
         assertEqual "" mempty actual
+    , testCase "Simplifies Implies" $ do
+        let expect =
+                Conditional
+                    { term = Mock.constr10 fOfX
+                    , predicate =
+                        makeAndPredicate
```

Comment:
It's not very important, but I wonder: why is the `\and` left-associated here?

Reply:
I think that there is no good reason for that, the previous code used foldl1' so I used the same convention.